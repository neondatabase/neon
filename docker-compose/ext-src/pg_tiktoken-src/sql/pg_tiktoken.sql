-- Load the extension
CREATE EXTENSION IF NOT EXISTS pg_tiktoken;

-- Test encoding function
SELECT tiktoken_encode('cl100k_base', 'Hello world!');

-- Test token count function
SELECT tiktoken_count('cl100k_base', 'Hello world!');

-- Test encoding function with a different model
SELECT tiktoken_encode('r50k_base', 'PostgreSQL is amazing!');

-- Test token count function with the same model
SELECT tiktoken_count('r50k_base', 'PostgreSQL is amazing!');

-- Edge cases: Empty string
SELECT tiktoken_encode('cl100k_base', '');
SELECT tiktoken_count('cl100k_base', '');

-- Edge cases: Long text
SELECT tiktoken_count('cl100k_base', repeat('word ', 100));

-- Edge case: Invalid encoding
SELECT tiktoken_encode('invalid_model', 'Test') AS should_fail;