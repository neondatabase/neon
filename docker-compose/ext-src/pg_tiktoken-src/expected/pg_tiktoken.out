-- Load the extension
CREATE EXTENSION IF NOT EXISTS pg_tiktoken;
-- Test encoding function
SELECT tiktoken_encode('cl100k_base', 'Hello world!');
 tiktoken_encode 
-----------------
 {9906,1917,0}
(1 row)

-- Test token count function
SELECT tiktoken_count('cl100k_base', 'Hello world!');
 tiktoken_count 
----------------
              3
(1 row)

-- Test encoding function with a different model
SELECT tiktoken_encode('r50k_base', 'PostgreSQL is amazing!');
     tiktoken_encode     
-------------------------
 {6307,47701,318,4998,0}
(1 row)

-- Test token count function with the same model
SELECT tiktoken_count('r50k_base', 'PostgreSQL is amazing!');
 tiktoken_count 
----------------
              5
(1 row)

-- Edge cases: Empty string
SELECT tiktoken_encode('cl100k_base', '');
 tiktoken_encode 
-----------------
 {}
(1 row)

SELECT tiktoken_count('cl100k_base', '');
 tiktoken_count 
----------------
              0
(1 row)

-- Edge cases: Long text
SELECT tiktoken_count('cl100k_base', repeat('word ', 100));
 tiktoken_count 
----------------
            101
(1 row)

-- Edge case: Invalid encoding
SELECT tiktoken_encode('invalid_model', 'Test') AS should_fail;
ERROR:  'invalid_model': unknown model or encoder
